internal/github/retry_roundtripper.go

```diff
@@ -0,0 +1,41 @@
+package github
+
+import (
+	"fmt"
+	"math"
+	"net/http"
+	"time"
+)
+
+type RetryTransport struct {
+	Transport http.RoundTripper
+	Retries   int
+}
+
+func (t *RetryTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	var (
+		resp *http.Response
+		err  error
+	)
+
+	for i := 0; i <= t.Retries; i++ {
+		resp, err = t.Transport.RoundTrip(req)
+		if err != nil {
+			fmt.Println("Error:", err)
+			continue
+		}
+
+		if resp.StatusCode >= 500 && resp.StatusCode < 600 {
+			fmt.Printf("Retrying request due to status code: %d\n", resp.StatusCode)
```

Comment:
Does merge gatekeeper use some logging framework already? If so, I think we should use it